core/src/main/java/org/apache/calcite/sql/validate/SqlValidatorImpl.java

```diff
@@ -3894,32 +3894,37 @@ protected void validateSelect(
 
     // Deduce which columns must be filtered.
     ns.mustFilterFields = ImmutableBitSet.of();
+    ns.mustFilterBypassFields = ImmutableBitSet.of();
     if (from != null) {
       final Set<SqlQualified> qualifieds = new LinkedHashSet<>();
```

Comment:
is `qualifieds` == `qualifiedFields` If so you could make it `qualifiedFields` and `qualifiedBypassFields` for consistency ?